ocp_resources/user_defined_network.py

```diff
@@ -66,3 +67,66 @@ def to_dict(self) -> None:
                 _spec["localNet"] = self.local_net
 
     # End of generated code
+
+
+class Layer2UserDefinedNetwork(UserDefinedNetwork):
+    """
+    UserDefinedNetwork layer2 object.
+
+    API reference:
+    https://ovn-kubernetes.io/api-reference/userdefinednetwork-api-spec/#layer2config
+    """
+
+    LAYER2: str = "Layer2"
+
+    def __init__(
+        self,
+        role: Optional[str] = None,
+        mtu: Optional[int] = None,
+        subnets: Optional[List[str]] = None,
+        join_subnets: Optional[List[str]] = None,
+        ipam_lifecycle: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        Create and manage UserDefinedNetwork with layer2 configuration
+
+        Args:
+            role (Optional[str]): role describes the network role in the pod.
```

Comment:
I would add something like "(primary or secondary)" at the end of the sentence, otherwise it's not clear what `role` might represent.

Reply:
please don't do `primary or secondary`, we do not want to be tied to API changes.
If in the next version, the API will also support `xxx` we will need to add this here.

Reply:
you have the API link in the class that should have explain what is every arg

Reply:
+1 @myakove

Reply:
> you have the API link in the class that should have explain what is every arg

Then why do we need the docstring at all?
If we add docstring, then it should be clear, otherwise it's useless.
You don't have to say that `role` is limited to "primary" and "secondary", because like Meni said - it can be changed or expanded in the future. You can add these as examples of possible values for `role`. The current explanation ("role describes the network role in the pod.") is not clear at all, and not helpful in understanding this field.
It's like you ask someone "what does 'name' mean?", and they answer "name is the name that you have".